preprocess_observation.py

```diff
@@ -21,25 +21,69 @@ def preprocess_observation_helper(mydict):
     for pokemon in mydict['side']['pokemon']:
         name = pokemon['ident'][4:]
         tempDict = {'Aggron': 0, 'Arceus': 1, 'Cacturne': 2, 'Dragonite': 3, 'Druddigon': 4, 'Uxie': 5}
-        health = int(pokemon['condition'].split('/')[0].split(' ')[0])
-        retval.append([809*2 + tempDict[name], health])
-
-        index = pokedex[name.lower()]['num']
-        if pokemon['active']:
-            retval.append([index-1, 1])
+        # Add health information for this pokemon
+        condition = pokemon['condition'].split('/')[0].split(' ')
+        health = int(condition[0])
+        retval.append([NUM_POKEMON*2 + tempDict[name], health])
+        # Add status information for this Pokemon
+        if len(condition) != 1 and condition[1] != 'fnt':
+            # in the future, a numerical value (e.g. 2 turns of sleep remaining) would be nice instead of just 1/0.
+            # TODO: confusion is not mutually exclusive with other status conditions, take account of this
+            assert(len(condition) == 2)
+            assert(condition[1] in STATUS_DICT)
+            for i in range(7):
+                retval.append([NUM_POKEMON*2 + TEAM_SIZE*2 + NUM_STATUS_CONDITIONS * tempDict[name] + i, STATUS_DICT[condition[1]] == i])
         else:
-            retval.append([index-1, 0])
+            for j in range(7):
+                retval.append([NUM_POKEMON*2 + TEAM_SIZE*2 + NUM_STATUS_CONDITIONS * tempDict[name] + j, 0])
+        # Add whether this Pokemon is active
+        index = pokedex[name.lower()]['num']
+        retval.append([index-1, pokemon['active']])
         retval2[tempDict[name]] = k
         k += 1
     other_indices = [pokedex['houndoom']['num'], pokedex['arceus']['num'], pokedex['cacturne']['num'], pokedex['dragonite']['num'], pokedex['druddigon']['num'], pokedex['uxie']['num']]
     assert(int(mydict['State'][6]) != 6)  # Remove this eventually
     cur_pokemon = int(mydict['State'][6])
-    for i in [0,1,2,3,4,5]:
+    for i in range(6):
```

Comment:
Make 6 a named constant.

Reply:
Not so sure on this one. There is code that cannot realistically import anything (e.g. it is written in TypeScript) that relies on this number being 6.